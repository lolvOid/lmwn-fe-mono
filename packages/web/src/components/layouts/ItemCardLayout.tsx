import { SyntheticEvent } from 'react';

interface ItemCardLayoutProps {
    onLayoutScroll?: (event: SyntheticEvent) => void;
    children: React.ReactNode;
}

const ItemCardLayout: React.FC<ItemCardLayoutProps> = ({ onLayoutScroll, children }) => {
    return (
        <>
            <div
                className="relative flex justify-center items-start h-[600px] overflow-y-auto max-w-[600px] py-8  bg-white lg:min-w-[200px] min-w-full mt-0 mx-auto my-0 z-50"
                onScroll={onLayoutScroll}
            >
                <div className="mx-0 lg:mx-auto my-0 w-full lg:w-auto px-4">
                    <div className="grid grid-cols-2 lg:grid-cols-2 gap-4 py-2">{children}</div>
                </div>
            </div>
        </>
    );
};

export default ItemCardLayout;
