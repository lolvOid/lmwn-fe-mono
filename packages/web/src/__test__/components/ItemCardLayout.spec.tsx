import { render, screen } from '@testing-library/react';
import ItemCardLayout from '@/components/layouts/ItemCardLayout';
import renderer from 'react-test-renderer';

describe('ItemCardLayout', () => {
    it('Render children properly', () => {
        render(
            <ItemCardLayout>
                <div>Div Component 1</div>
                <div>Div Component 2</div>
            </ItemCardLayout>,
        );

        expect(screen.getByText('Div Component 1')).toBeInTheDocument();
        expect(screen.getByText('Div Component 2')).toBeInTheDocument();
    });

    it('Apply styles correctly', () => {
        render(
            <ItemCardLayout>
                <div>Div Component 1</div>
            </ItemCardLayout>,
        );

        const layout = screen.getByText('Div Component 1').closest('.relative');
        const container = screen.getByText('Div Component 1').closest('.flex');

        expect(layout).toHaveClass(
            'h-[600px] overflow-y-auto max-w-[600px] py-8 bg-white lg:min-w-[200px] min-w-full mt-0 mx-auto my-0 z-50',
        );
        expect(container).toHaveClass('flex justify-center items-start');
    });

    it('Should render with different number of columns on lg screens', () => {
        render(
            <ItemCardLayout>
                <div>Div Component 1</div>
                <div>Div Component 2</div>
                <div>Div Component 3</div>
            </ItemCardLayout>,
        );

        const grid = screen.getByText('Div Component 1').closest('.grid');

        expect(grid).toHaveClass('grid-cols-2 lg:grid-cols-2 gap-4 py-2');
    });

    it('Should render and match the default snapshot', () => {
        const layout = renderer
            .create(
                <ItemCardLayout>
                    <div>Div Component 1</div>
                </ItemCardLayout>,
            )
            .toJSON();

        expect(layout).toMatchSnapshot();
    });
});
