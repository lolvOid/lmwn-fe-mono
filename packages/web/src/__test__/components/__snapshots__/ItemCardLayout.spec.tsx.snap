// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCardLayout Should render and match the default snapshot 1`] = `
<div
  className="relative flex justify-center items-start h-[600px] overflow-y-auto max-w-[600px] py-8  bg-white lg:min-w-[200px] min-w-full mt-0 mx-auto my-0 z-50"
>
  <div
    className="mx-0 lg:mx-auto my-0 w-full lg:w-auto px-4"
  >
    <div
      className="grid grid-cols-2 lg:grid-cols-2 gap-4 py-2"
    >
      <div>
        Div Component 1
      </div>
    </div>
  </div>
</div>
`;
